sdk/src/client/secret/ledger_nano.rs

```diff
@@ -400,24 +393,19 @@ impl SecretManagerConfig for LedgerSecretManager {
 /// If criteria are not met, blind signing is needed.
 /// This method finds out if we have to switch to blind signing mode.
 pub fn needs_blind_signing(prepared_transaction: &PreparedTransactionData, buffer_size: usize) -> bool {
-    match &prepared_transaction.essence {
-        TransactionEssence::Regular(essence) => {
-            for output in essence.outputs().iter() {
-                // only basic outputs allowed
-                if let Output::Basic(output) = output {
-                    if output.simple_deposit_address().is_some() {
-                        continue;
-                    }
-                }
-                // not fine, return
-                return true;
-            }
-        }
+    let TransactionEssence::Regular(essence) = &prepared_transaction.essence;
+
+    if essence
+        .outputs()
+        .iter()
+        .any(|output| !matches!(output, Output::Basic(o) if o.simple_deposit_address().is_some()))
```

Comment:
ah ... you prefer this version ... any specific reason why?

Reply:
No, I just glanced at your comment before writing it so I happened to do the opposite. They should be functionally identical.

Reply:
yeah, they are. Had this one, too, but then switched to `.all`.